// metro4/Update_Builder.js
module.exports = function(grunt) {

    "use strict";

    require('load-grunt-tasks')(grunt);

    grunt.initConfig({
        pkg: grunt.file.readJSON('package.json'),

        ftp_push: {
            upd_vendors: {
                options: {
                    authKey: "metroui",
                    host: "eg77.mirohost.net",
                    dest: "builder.metroui.org.ua/vendors/metro/",
                    port: 21,
                    incrementalUpdates: false
                },
                files: [
                    {
                        expand: true,
                        cwd: "build",
                        src: ['**/*']
                    }
                ]
            },
            upd_less: {
                options: {
                    authKey: "metroui",
                    host: "eg77.mirohost.net",
                    dest: "builder.metroui.org.ua/source/less/",
                    port: 21,
                    incrementalUpdates: false
                },
                files: [
                    {
                        expand: true,
                        cwd: "less",
                        src: ['**/*']
                    }
                ]
            },
            upd_js: {
                options: {
                    authKey: "metroui",
                    host: "eg77.mirohost.net",
                    dest: "builder.metroui.org.ua/source/js/",
                    port: 21,
                    incrementalUpdates: false
                },
                files: [
                    {
                        expand: true,
                        cwd: "js",
                        src: ['**/*']
                    }
                ]
            },
            upd_icons: {
                options: {
                    authKey: "metroui",
                    host: "eg77.mirohost.net",
                    dest: "builder.metroui.org.ua/source/icons/",
                    port: 21,
                    incrementalUpdates: false
                },
                files: [
                    {
                        expand: true,
                        cwd: "icons",
                        src: ['**/*']
                    }
                ]
            },
            upd_package: {
                options: {
                    authKey: "metroui",
                    host: "eg77.mirohost.net",
                    dest: "builder.metroui.org.ua/",
                    port: 21,
                    incrementalUpdates: false
                },
                files: [
                    {
                        expand: true,
                        cwd: '.',
                        src: ['package.json']
                    }
                ]
            },
            upd_config: {
                options: {
                    authKey: "metroui",
                    host: "eg77.mirohost.net",
                    dest: "builder.metroui.org.ua/",
                    port: 21,
                    incrementalUpdates: false
                },
                files: [
                    {
                        expand: true,
                        cwd: 'builder',
                        src: ['config.json']
                    }
                ]
            }
        }
    });

    grunt.registerTask('default', [
        'ftp_push'
    ]);

};
